src/ioctls/device.rs

```diff
@@ -110,4 +104,50 @@ mod tests {
         assert!(device_fd.set_device_attr(&dist_attr).is_err());
         assert_eq!(errno::Error::last().errno(), 25);
     }
+
+    #[test]
+    #[cfg(target_arch = "aarch64")]
+    fn test_create_device() {
+        use kvm_bindings::kvm_device_type_KVM_DEV_TYPE_FSL_MPIC_20;
+
+        let kvm = Kvm::new().unwrap();
+        let vm = kvm.create_vm().unwrap();
+
+        let mut gic_device = kvm_bindings::kvm_create_device {
+            type_: kvm_device_type_KVM_DEV_TYPE_FSL_MPIC_20,
+            fd: 0,
+            flags: KVM_CREATE_DEVICE_TEST,
+        };
+        // This fails on aarch64 as it does not use MPIC (MultiProcessor Interrupt Controller), it uses
+        // the VGIC.
+        assert!(vm.create_device(&mut gic_device).is_err());
+
+        gic_device.type_ = kvm_device_type_KVM_DEV_TYPE_ARM_VGIC_V3;
+
+        let device_fd = vm.create_device(&mut gic_device);
+        let device_fd = match device_fd {
+            Ok(fd) => fd,
+            Err(_) => {
+                gic_device.type_ = kvm_device_type_KVM_DEV_TYPE_ARM_VGIC_V2;
+                vm.create_device(&mut gic_device)
+                    .expect("Cannot create KVM device")
+            }
+        };
+
+        let raw_fd = unsafe { libc::dup(device_fd.as_raw_fd()) };
+        assert!(raw_fd >= 0);
+        let device_fd = unsafe { DeviceFd::from_raw_fd(raw_fd) };
```

Comment:
Why do this? `device_fd` is already a `DeviceFd`. Code should be equivalent without these 3 lines playing with the `unsafe` underlying `RawFd`.

Reply:
Yes, these 3 lines are unnecessary. I introduced them when splitting from the old test function. :)
In x86 test case, the `raw fd` was not needed either. I deleted them as well.